frontend/model/contracts/shared/nativeNotification.js

```diff
@@ -9,20 +10,39 @@ export async function requestNotificationPermission (force: boolean = false): Pr
     (force && Notification.permission === 'granted')) {
     return null
   }
-
   try {
-    return await Notification.requestPermission()
+    const permission = await Notification.requestPermission()
+    if (permission === 'granted') {
+      sbp('state/vuex/commit', 'setNotificationGranted', true)
+    } else {
+      sbp('state/vuex/commit', 'setNotificationGranted', false)
+    }
+    return permission
   } catch (e) {
     return null
   }
 }
 
-export function makeNotification ({ title, body, icon }: {
-  title: string, body: string, icon: string
+export function makeNotification ({ title, body, icon, url }: {
+  title: string, body: string, icon?: string, url?: string
 }): void {
   if (typeof Notification === 'undefined' || Notification.permission !== 'granted') {
     return
   }
-  // eslint-disable-next-line
-  new Notification(title, { body, icon })
+
+  const notification = new Notification(title, { body, icon })
+  if (url) {
+    notification.onclick = function (event) {
+      event.preventDefault()
+      open(url, '_blank')
```

Comment:
I don't feel really comfortable with the `open()` function. Wouldn't it be better to somehow push the URL onto the Vue router?

Reply:
Yeah. I am not sure why I didn't think that way.